subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/LinkBundle.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.tasks;
+
+import org.gradle.api.Incubating;
+import org.gradle.nativeplatform.internal.BundleLinkerSpec;
+import org.gradle.nativeplatform.internal.DefaultLinkerSpec;
+import org.gradle.nativeplatform.internal.LinkerSpec;
+
+/**
+ * Links a binary bundle from object files and imported libraries.
+ *
+ * @since 4.3
+ */
+@Incubating
+public class LinkBundle extends AbstractLinkTask {
```

Comment:
We should rename this to something like `LinkMachOBundle` to make it clear that this isn't creating the other kind of 'bundle'

Reply:
Good point, done.